[clang-tidy][modernize-use-starts-ends-with] Fix minor mistake in error message

nicovank opened the pull request:
In one of the cases recently added to this check in #110448, the error message is no longer accurate as the comparison is not with zero. #116033 brought my attention to this as it may add another comparison without zero.

Remove the `[!=] 0` part of the diagnostic. This is in line with some other checks e.g. modernize-use-emplace.

```
> cat tmp.cpp
#include <string>
bool f(std::string u, std::string v) {
  return u.rfind(v) == u.size() - v.size();
}

# Before.
> ./build/bin/clang-tidy -checks="-*,modernize-use-starts-ends-with" tmp.cpp -- -std=c++20
tmp.cpp:3:12: warning: use ends_with instead of rfind() == 0 [modernize-use-starts-ends-with]
    3 |   return u.rfind(v) == u.size() - v.size();
      |            ^~~~~~ ~~~~~~~~~~~~~~~~~~~~~~~~
      |            ends_with( )

# After.
> ./build/bin/clang-tidy -checks="-*,modernize-use-starts-ends-with" tmp.cpp -- -std=c++20
tmp.cpp:3:12: warning: use ends_with instead of rfind [modernize-use-starts-ends-with]
    3 |   return u.rfind(v) == u.size() - v.size();
      |            ^~~~~~ ~~~~~~~~~~~~~~~~~~~~~~~~
      |            ends_with( )
```

5chmidti commented on clang-tools-extra/docs/ReleaseNotes.rst:
```diff
@@ -244,7 +244,7 @@ Changes in existing checks
 - Improved :doc:`modernize-use-starts-ends-with
   <clang-tidy/checks/modernize/use-starts-ends-with>` check to handle two cases
-  that can be replaced with ``ends_with``
+  that can be replaced with ``ends_with``. Minor change to error message.
```
Something like `and a small adjustment to a diagnostic message` sounds better

5chmidti commented on clang-tools-extra/test/clang-tidy/checkers/modernize/use-starts-ends-with.cpp:
Please also add a test for the case you're doing this for, i.e., one without `[!=]= 0`, so that these don't regress in the future

nicovank committed:
[clang-tidy][modernize-use-starts-ends-with] Fix minor mistake in error message

In one of the cases recently added to this check in #110448, the error message is no longer accurate as the comparison is not with zero. #116033 brought my attention to this as it may add another comparison without zero.

Remove the `[!=] 0` part of the diagnostic. This is in line with some other checks e.g. modernize-use-emplace.

```
> cat tmp.cpp
#include <string>
bool f(std::string u, std::string v) {
  return u.rfind(v) == u.size() - v.size();
}

# Before.
> ./build/bin/clang-tidy -checks="-*,modernize-use-starts-ends-with" tmp.cpp -- -std=c++20
tmp.cpp:3:12: warning: use ends_with instead of rfind() == 0 [modernize-use-starts-ends-with]
    3 |   return u.rfind(v) == u.size() - v.size();
      |            ^~~~~~ ~~~~~~~~~~~~~~~~~~~~~~~~
      |            ends_with( )

# After.
> ./build/bin/clang-tidy -checks="-*,modernize-use-starts-ends-with" tmp.cpp -- -std=c++20
tmp.cpp:3:12: warning: use ends_with instead of rfind [modernize-use-starts-ends-with]
    3 |   return u.rfind(v) == u.size() - v.size();
      |            ^~~~~~ ~~~~~~~~~~~~~~~~~~~~~~~~
      |            ends_with( )
```

nicovank commented:
Incorporated in #116033.
